ThunderTable/ScrollOffsetManagable.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  ScrollDelegatable.swift
+//  ThunderTable
+//
+//  Created by Simon Mitchell on 02/09/2020.
+//  Copyright © 2020 threesidedcube. All rights reserved.
+//
+
+import UIKit
+
+/// A simpler version of `UIScrollViewDelegate` so we don't intefere
+/// with users `UIScrollViewDelegate` implementations
+public protocol ScrollOffsetDelegate: class {
+    
+    /// Called when the content offset of the scroll view changes due to `scrollViewDidScroll`
+    /// - Parameters:
+    ///   - scrollable: The scrollable that the change was for
+    func scrollViewDidChangeContentOffset(_ scrollable: ScrollOffsetManagable)
+}
+
+/// A protocol implemented to allow control and listening to scroll view offset changes
+/// by `ScrollOfffsetManager`
+public protocol ScrollOffsetManagable: class {
```

Comment:
Do we want to just "inherit" protocol `ScrollOffsetDelegate`. Actually not sure if inherit is the correct term here